src/modules/vtol_att_control/vtol_att_control_main.cpp

```diff
@@ -542,12 +548,62 @@ VtolAttitudeControl::Run()
 
 		// Advertise/Publish vtol vehicle status
 		_vtol_vehicle_status.timestamp = hrt_absolute_time();
+		_vtol_vehicle_status.mc_weights[0] = _vtol_type->_mc_roll_weight;
+		_vtol_vehicle_status.mc_weights[1] = _vtol_type->_mc_pitch_weight;
+		_vtol_vehicle_status.mc_weights[2] = _vtol_type->_mc_yaw_weight;
+		_vtol_vehicle_status.mc_weights[3] = _vtol_type->_mc_throttle_weight;
 		_vtol_vehicle_status_pub.publish(_vtol_vehicle_status);
 	}
 
 	perf_end(_loop_perf);
 }
 
+void VtolAttitudeControl::update_filtered_airspeed(const struct airspeed_validated_s &sample)
+{
+	if (!sample.airspeed_sensor_measurement_valid || !PX4_ISFINITE(sample.calibrated_airspeed_m_s)) {
+		if (_last_airspeed_index != -1) {
+			PX4_WARN("Received invalied calibrated airspeed");
+		}
+
+		_last_airspeed_index = -1;
+		return;
+	}
+
+	if (sample.selected_airspeed_index != _last_airspeed_index) {
+		// First sample from a new airspeed sensor. Reset the filter.
+		_airspeed_filtered.reset(sample.calibrated_airspeed_m_s);
+		_last_airspeed_timestamp = sample.timestamp;
+		_last_airspeed_index = sample.selected_airspeed_index;
+		return;
+	}
+
+	float dt = static_cast<float>(sample.timestamp - _last_airspeed_timestamp);
+
+	if (dt >= 0) {
+		if (_last_airspeed_index != -1) {
+			PX4_WARN("Airspeed timestamp not monotonic");
+		}
+
+		_last_airspeed_index = -1;
+		return;
+	}
+
+	// We update the filter sample rate to account for varying sample rates
+	_airspeed_filtered.setParameters(dt, _params.vt_aspd_tau);
+	_airspeed_filtered.update(sample.calibrated_airspeed_m_s);
+	_last_airspeed_timestamp = sample.timestamp;
+	_last_airspeed_index = sample.selected_airspeed_index;
+}
```

Comment:
```suggestion
	if (!sample.airspeed_sensor_measurement_valid || !PX4_ISFINITE(sample.calibrated_airspeed_m_s)) {
			_airspeed_filtered.update(sample.calibrated_airspeed_m_s);
	}
}
```

Reply:
To understand why the extra complexity it worthy it: Why won't my suggested simplification above work?

Reply:
We can skip the resets, since a long dt basically equals a reset.
However, we don't know in advance the airspeed sensor's sample rate, so must update the filter parameters.
Could simplify to this

```
	if (!sample.airspeed_sensor_measurement_valid || !PX4_ISFINITE(sample.calibrated_airspeed_m_s)) {
		if (!_airspeed_filtered_valid) {
			PX4_WARN("Received invalid calibrated airspeed");
		}
		_airspeed_filtered_valid = false;
		return;
	}

	if (sample.timestamp <= _airspeed_filtered_last_timestamp) {
		if (_airspeed_filtered_valid) {
			PX4_WARN("Airspeed timestamp not monotonic");
		}

		_airspeed_filtered_valid = false;
		return;
	}

	// We must update the filter parameters, since we don't know the sensor sample rate
	float dt = static_cast<float>(sample.timestamp - _airspeed_filtered_last_timestamp);
	_airspeed_filtered.setParameters(dt, _params.vt_aspd_tau);
	_airspeed_filtered.update(sample.calibrated_airspeed_m_s);
	_airspeed_filtered_last_timestamp = sample.timestamp;
	_airspeed_filtered_valid = true;
```

Reply:
We also need to output invalid data out if our input is invalid, since we want to trigger failsafes in the transition code if we loose airspeed.